File: tools/docs-generator/lib/level/utility/stage/broker/message/message_prefab_xml_reader.py
import xml.etree.ElementTree as EXml

from lib.level.utility.stage.broker.message.message_prefab import MessagePrefab


class MessagePrefabXmlReader(object):
    MESSAGE_ELEMENT = "message"
    TIME_ELEMENT = "time"
    TRANSLATION_ENTRY_KEY_ELEMENT = "entryKey"
    FACE_ID_ELEMENT = "faceId"

    def read_prefab_from_string(self, xml_string):
        try:
            element = EXml.fromstring(xml_string)
            time = float(element.find(self.TIME_ELEMENT).text.strip())
            translation_entry_key = str(element.find(self.TRANSLATION_ENTRY_KEY_ELEMENT).text.strip())
            face_id = str(element.find(self.FACE_ID_ELEMENT).text.strip())

            return MessagePrefab(time, translation_entry_key, face_id)
        except Exception as e:
            raise MessagePrefabXmlReadException(xml_string, e)


class MessagePrefabXmlReadException(Exception):
    MESSAGE_TEMPLATE = "Cannot read message prefab from xml string {}. Cause: {}."

    def __init__(self, xml_string, cause):
        super().__init__(self.MESSAGE_TEMPLATE.format(xml_string, cause))

File: core/src/remove/tanks/game/level/utility/stage/broker/message/MessagePrefabXmlReader.java
package remove.tanks.game.level.utility.stage.broker.message;

import com.badlogic.gdx.utils.XmlReader;

/**
 * @author Mateusz Długosz
 */
public final class MessagePrefabXmlReader {
    public static final String MESSAGE_ELEMENT = "message";

    private static final String TIME_ELEMENT = "time";
    private static final String ENTRY_KEY_ELEMENT = "entryKey";
    private static final String FACE_ID_ELEMENT = "faceId";

    public MessagePrefab readMessagePrefab(XmlReader.Element element) {
        try {
            return new MessagePrefab(
                    readTime(element),
                    readEntryKey(element),
                    readFaceId(element)
            );
        } catch (Exception e) {
            throw new MessagePrefabXmlReadException(element, e);
        }
    }

    private float readTime(XmlReader.Element element) {
        return Float.valueOf(element.getChildByName(TIME_ELEMENT).getText().trim());
    }

    private String readEntryKey(XmlReader.Element element) {
        return element.getChildByName(ENTRY_KEY_ELEMENT).getText().trim();
    }

    private String readFaceId(XmlReader.Element element) {
        if (element.getChildByName(FACE_ID_ELEMENT) == null) {
            return null;
        }
        return element.getChildByName(FACE_ID_ELEMENT).getText().trim();
    }
}
